src/Dapr.Client/DaprClientGrpc.cs

```diff
@@ -1373,6 +1375,475 @@ private Autogenerated.StateItem ToAutogeneratedStateItem(StateTransactionRequest
             var resp = await client.UnsubscribeConfigurationAsync(request, options);
             return new UnsubscribeConfigurationResponse(resp.Ok, resp.Message);
         }
+
+        #endregion
+
+        #region Cryptography
+
+        private async Task<byte[]> EncryptStreamAsync(string vaultResourceName, Stream stream,
+            KeyWrapAlgorithm algorithm, string keyName, DataEncryptionCipher dataEncryptionCipher,
+            string decryptionKeyName, bool omitDecryptionKey, CancellationToken cancellationToken = default)
+        {
+            ArgumentVerifier.ThrowIfNullOrEmpty(vaultResourceName, nameof(vaultResourceName));
+            ArgumentVerifier.ThrowIfNullOrEmpty(keyName, nameof(keyName));
+            ArgumentVerifier.ThrowIfNull(stream, nameof(stream));
+
+            var encryptRequestOptions = new Autogenerated.EncryptRequestOptions
+            {
+                ComponentName = vaultResourceName,
+                DataEncryptionCipher = dataEncryptionCipher.GetValueFromEnumMember(),
+                KeyName = keyName,
+                KeyWrapAlgorithm = algorithm.GetValueFromEnumMember(),
+                OmitDecryptionKeyName = omitDecryptionKey
+            };
+
+            if (!omitDecryptionKey)
+            {
+                ArgumentVerifier.ThrowIfNullOrEmpty(decryptionKeyName, nameof(decryptionKeyName));
+                encryptRequestOptions.DecryptionKeyName = decryptionKeyName;
+            }
+
+            var options = CreateCallOptions(headers: null, cancellationToken);
+            var duplexStream = client.EncryptAlpha1(options);
+
+            //Start with passing the metadata about the encryption request itself in the first message
+            await duplexStream.RequestStream.WriteAsync(
+                new Autogenerated.EncryptRequest { Options = encryptRequestOptions }, cancellationToken);
+
+            //Send the plaintext bytes in 4 kb blocks in subsequent messages
+            const int blockSize = 4 * 1024; // 4 kb
+            await using (var bufferedStream = new BufferedStream(stream, blockSize))
```

Comment:
Maybe this is because my knowledge of C# is limited, but if I understand correctly here you first send the entire data to Dapr in chunks, and then (when the data has been fully sent), you start reading the response.

Ideally, you'd be reading the response _at the same time_ as you're sending the data to Dapr.

The idea is that you can send a chunk to Dapr, and then Dapr sends it back to you encrypted/decrypted as soon as it's ready

_(However, don't wait for each chunk; just write and read in parallel)_

Reply:
This is a good catch, as otherwise the request is getting implicitly buffered in the runtime.  I recall having to do something similar in the Dapr .NET pluggable components SDK to support the pub-sub and binding streaming calls.

Reply:
Great catch - updated in the latest commit for both encrypt and decrypt.